lib/tasks/db.rake

```diff
@@ -30,45 +30,153 @@ task 'db:force_skip_persist' do
   GlobalSetting.skip_redis = true
 end
 
-task 'db:create' => [:load_config] do |_, args|
-  if MultisiteTestHelpers.create_multisite?
-    unless system("RAILS_ENV=test RAILS_DB=discourse_test_multisite rake db:create")
+def config_to_url(config)
+  if config[:username] || config[:password]
+    userinfo = [config[:username], config[:password]].join(":")
+  end
 
-      STDERR.puts "-" * 80
-      STDERR.puts "ERROR: Could not create multisite DB. A common cause of this is a plugin"
-      STDERR.puts "checking the column structure when initializing, which raises an error."
-      STDERR.puts "-" * 80
-      raise "Could not initialize discourse_test_multisite"
-    end
+  URI::Generic.new(
+    config[:adapter],
+    userinfo,
+    config[:hostname] || "localhost",
+    config[:port],
+    nil,
+    "/#{config[:database]}",
+    nil,
+    nil,
+    nil
+  ).to_s
+end
+
+# db:create and related tasks
+
+task "multisite:create" => ["db:load_config"] do
+  next if !Rails.env.development? || !ENV["RAILS_DB"] || ENV["DATABASE_URL"]
+
+  spec = RailsMultisite::ConnectionManagement.connection_spec(db: ENV["RAILS_DB"])
+  database_url = config_to_url(spec.config)
+  system("DATABASE_URL=#{database_url} rake db:create")
```

Comment:
I think `system("RAILS_DB=<db name> rake db:create")` will work and there should not be a need for us to have to generate the database url.

Reply:
It didn't really! That was one of the reasons for this PR. There was no code (in core or rails_multisite) that would handle db:create in multisite.

It worked only by accident when your site had the same name as its database, because of how our **test** db is set up in config/database.yml

But, for example, when your multisite config looked like this (as `multisite.yml.production-sample` and rails_multisite readme suggest)

```yml
starwars:
  database: discourse_starwars
```

running `RAILS_DB=starwars rake db:create` resulted in creating dbs `discourse_development` and `starwars` (incorrect)

running `RAILS_DB=discourse_starwars rake db:create` resulted in error: `ERROR: discourse_starwars not found!`

Reply:
O icic. I guess we don't encounter this problem in production because we expect the database to already exist. Can this be a feature or fix that goes into `rails_multisite`?